src/js/components/Button/Button.js

```diff
@@ -92,12 +96,80 @@ const getIconColor = (paths = [], theme, colorProp, kind) => {
   return result[1] || undefined;
 };
 
+const getBadgeDimension = (dimension, badgeProp, badgeRef, theme) => {
+  if (
+    typeof badgeProp === 'number' ||
+    (typeof badgeProp === 'object' && badgeProp.value)
+  ) {
+    const borderWidth = badgeProp.border
+      ? parseInt(
+          theme.global.borderSize[badgeProp.border.size].replace('px', ''),
+          10,
+        ) * 2
```

Comment:
Why the `* 2` here?

Reply:
Border width applies on both left and right

Reply:
wondering if either a comment or renaming "borderWidth" might help. "widthUsedByBorders" is a bit long, but gets at it